### PR TITLE
Arsenal - Fixed stats & actions appearing when they shouldn't be

### PR DESCRIPTION
**Pull request by johnb432**
**When merged this pull request will:**
- Addresses #9722.
- Some cleanup.

### IMPORTANT

- If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: addons/arsenal/XEH_preInit.sqf [hunk 1]
```diff
@@ -15,55 +15,61 @@ PREP_RECOMPILE_END;
 [QGVAR(statsToggle), {
     params ["_display", "_showStats"];
 
+    _showStats = GVAR(showStats) && _showStats;
```
Comment (johnb432) [inline addons/arsenal/XEH_preInit.sqf:18]: `GVAR(showStats)` isn't even used - I want to remove it, but was wondering if there are any objections.

File: addons/arsenal/XEH_preInit.sqf [hunk 2]
```diff
@@ -15,55 +15,61 @@ PREP_RECOMPILE_END;
...
         _actionsPreviousPageCtrl,
         _actionsNextPageCtrl,
         _actionsCurrentPageCtrl
     ];
+
+    if (!_showActions) exitWith {};
+
+    GVAR(actionsInfo) params ["_control", "_curSel", "_itemCfg"];
+
+    [QGVAR(displayActions), [_display, _control, _curSel, _itemCfg]] call CBA_fnc_localEvent;
```
Comment (johnb432) [inline addons/arsenal/XEH_preInit.sqf:67]: Not a huge fan of this solution, as I feel it could fail.

File: addons/arsenal/XEH_preInit.sqf (hunk 1 above)
Comment (LinkIsGrim) [reply addons/arsenal/XEH_preInit.sqf:18]: @BrettMayson do you use that at all?

File: addons/arsenal/XEH_preInit.sqf (hunk 2 above)
Comment (LinkIsGrim) [reply addons/arsenal/XEH_preInit.sqf:67]: How so? Isn't this just doing the same thing as selecting a different item/switching tabs would?

Comment (johnb432) [reply addons/arsenal/XEH_preInit.sqf:67]: I'm wondering: If the arsenal refreshes and removes the currently selected item, what happens?

**Comment by johnb432** [comment]
https://github.com/acemod/ACE3/blob/5903e561a99787865c265c23fa84fbb5e889fbed/addons/arsenal/functions/fnc_buttonHide.sqf#L55-L57

https://github.com/acemod/ACE3/blob/5903e561a99787865c265c23fa84fbb5e889fbed/addons/arsenal/XEH_preInit.sqf#L19-L21

Something is redundant, my guess is that we forgot to remove the IDCs from `FUNC(buttonhide)`.

File: addons/arsenal/XEH_preInit.sqf (hunk 1 above)
Comment (BrettMayson) [reply addons/arsenal/XEH_preInit.sqf:18]: I do not

**Comment by johnb432** [comment]
> Something is redundant, my guess is that we forgot to remove the IDCs from `FUNC(buttonhide)`.
> 
> I'll be updating that.

Decided to remove the events instead, as they serve little purpose.
I can add them back, but I see little incentive, as they weren't public. You can just use the `QGVAR(displayStats)` and `QGVAR(displayActions)` if you want anyway.

**Review by LinkIsGrim — APPROVED** [review]

**Pull request merged**